code/game/objects/items.dm

```diff
@@ -167,7 +167,8 @@ GLOBAL_DATUM_INIT(fire_overlay, /image, image("icon" = 'icons/goonstation/effect
 		m.unEquip(src, 1)
 	else if(istype(loc, /obj/item/storage))
 		var/obj/item/storage/S = loc
-		S.remove_from_storage(src)
+		if(!QDELETED(S)) // Can happen when the storage gets deleted, thus deleting the contents
+			S.remove_from_storage(src) // Not required to call. Will only cause runtimes
```

Comment:
Dropping the contents will bring various functional changes with it as well. For example when a bag gets destroyed by acid.
I'd be fine with that. But it won't fix the issue here. Since the storage in question will already be deleted before the items inside get deleted due to how the list is constructed.
It indeed is a race condition like situation where you can not be sure if the item gets deleted first. Or the storage gets deleted first.